### ISSUE TITLE
Fuse.Share API slightly broken on Android

### ISSUE THREAD

**Issue by eksperts** [fusetools/fuselibs-public#255]
[Reported on community](https://www.fusetools.com/community/forums/howto_discussions/share_a_local_image?page=1&highlight=5816d2d2-c8ff-46c0-8788-fabf55e9677a#post-5816d2d2-c8ff-46c0-8788-fabf55e9677a).

Found an [answer on SO](https://stackoverflow.com/questions/26883259/gmail-5-0-app-fails-with-permission-denied-for-the-attachment-when-it-receives) that suggests doing: `replace startActivity with startActivityForResult`, otherwise we're getting "Permission denied for the attachment" when trying to share to Google apps at least.

Repro:
```
<App>
	<JavaScript>
		var Bundle = require("FuseJS/Bundle");
		var Share = require("FuseJS/Share");
		var FileSystem = require("FuseJS/FileSystem");
		function share() {
			console.log("share clicked");
			Bundle.extract('1.jpg', FileSystem.dataDirectory + '/1.jpg')
			.then(function(resultPath) {
				console.log(resultPath);
				Share.shareFile(resultPath, "image/jpg", "Message");
			});
		}
		module.exports = {
			share: share
		};
	</JavaScript>
	<Panel Width="240" Height="56" Clicked="{share}">
		<Text Value="Share" Alignment="Center" Color="#fff" />
		<Rectangle Color="#18f" CornerRadius="2" />
	</Panel>
</App>
```
Make sure you `:Bundle` a "1.jpg" file with it.

**Comment by fusebuild** [comment]
The forum thread [Share a local image](https://www.fusetools.com/community/forums/howto_discussions/share_a_local_image) posted by Cristian Karmy was linked to this issue.

**Comment by cbaggers** [comment]
@eksperts `startActivityForResult` didnt fix it in this case. Reading the SO post again it's the fact it's Bundled that is the problem. It's also a kind of fundamental problem as we cant just copy the file to the public storage for the user as android doesnt give us a callback to tell us when they are done with the file so we dont know when to delete it. Also putting data in the public area is a security issue so we wouldnt do that anyway.

It's a bugger, do you have any ideas?

**Comment by eksperts** [comment]
@cbaggers but we're extracting the file from the bundle and saving it in `FileSystem.dataDirectory`. Ain't that enough, or am I missing something?

**Comment by cbaggers** [comment]
@eksperts yeah it should be. Will test this again. Seems I minimized the test case too far :|

**Comment by cbaggers** [comment]
@eksperts Ah I should have seen this sooner bug https://www.fusetools.com/docs/fuse/filesystem/filesystemmodule/getdatadirectory this is the private directory for the app, so the same issue as before applies. Gmail doesnt have access to that file.

**Comment by eksperts** [comment]
@cbaggers okay, then as I read it the right solution (provided that's possible) would be to copy the file to a temporary globally-accessible location, and sharing it from there? And then perhaps cleaning up after ourselves.

**Comment by cbaggers** [comment]
@eksperts the problem is that the share api doesnt appear to tell us when it is done, we can't delete the image until it's sent or the email deleted so theres no hook to delete the file. And without that we would be littering :)

Also unless the user explicitly says that its ok to write to the public area I'd really try to avoid that, I think a user would rightly assume we are keeping their image private.
